internal/daemon/api_files.go

```diff
@@ -0,0 +1,576 @@
+// Copyright (c) 2021 Canonical Ltd
+//
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License version 3 as
+// published by the Free Software Foundation.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+//
+// You should have received a copy of the GNU General Public License
+// along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+package daemon
+
+import (
+	"encoding/json"
+	"errors"
+	"fmt"
+	"io"
+	"io/ioutil"
+	"mime"
+	"mime/multipart"
+	"net/http"
+	"net/textproto"
+	"os"
+	"os/user"
+	pathpkg "path"
+	"strconv"
+	"time"
+
+	"github.com/canonical/pebble/internal/osutil"
+	"github.com/canonical/pebble/internal/osutil/sys"
+)
+
+const minBoundaryLength = 8
```

Comment:
Aren't boundaries pretty long strings in general?  Can we do at least 32 here, considering Python and Go conventions?

Reply:
Good call. I'd originally used 8 so I didn't have to go crazy in the tests, but now I'm just using `01234567890123456789012345678901` is the tests, no big deal.